app/model/build.js

```diff
@@ -1,13 +1,73 @@
+// Imports for packages
 import mongoose from 'mongoose';
+import Promise from 'bluebird';
 
+// Local imports
+import app from '~/';
+
+// Mongoose schema for builds
 const BuildSchema = mongoose.Schema({
   arch:       String,
   target:     String,
   version:    String,
-  started:    Date,
+  started: {
+    type:     Date,
+    default:  Date.now(),
+  },
   finished:   Date,
-  status:     String,
+  status: {
+    type:     String,
+    default: 'QUEUED',
+  },
   log:        String,
 });
 
+// Saves application with updated build information
+// Returns promise of saved application
+BuildSchema.methods.nestSave = function() {
```

Comment:
huh, oh well. at a node-stack startup i was working at a couple years ago, one of our technical advisors suggested we might write our own abstract/relatively agnostic database layer on top of mongoose, but that could be higher level and wrap around these kinds of warts (and, moreover, provide another level of isolation/abstraction from the actual database in case we ever switched)

almost certainly overkill at this stage, but something to keep in the back of the mind long-term I guess